src/scripts/activate-user.ts

```diff
@@ -0,0 +1,197 @@
+ * @returns a TaskEither that resolves to the user if the user exists, otherwise an error message.
+ */
+const validateUserExist = (
+  service: UserService,
+  userId: string
+): TE.TaskEither<string, User> =>
+  pipe(
+    userId,
+    service.getUser,
+    TE.chain(
+      flow(
+        O.fold(
+          () => TE.left(`User [${userId}] does not exist. Can not activate.`),
+          // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+          (user) => TE.right(user)
+        )
+      )
+    )
+  );
+
+/**
+ * Checks to see if user has a status of staged or deprovisioned.
+ * @param user - the user to check the status of.
+ * @returns - a TaskEither that resolves to the user if the user has a status of staged or deprovisioned, otherwise an error message.
+ */
+const validateUserStatusPriorToActivation = (
+  user: User
+): TE.TaskEither<string, ActivatableUser> => {
+  const userStatus = user.status;
+  const activeStatus = okta.UserStatus.ACTIVE;
+  const context = `User [${user.id}] [${user.email}] has status [${userStatus}]. Activation is reserved for users with status ${okta.UserStatus.STAGED} or ${okta.UserStatus.DEPROVISIONED}.`;
+
+  // eslint-disable-next-line sonarjs/no-small-switch, functional/no-conditional-statement
+  switch (userStatus) {
+    case okta.UserStatus.ACTIVE: {
+      return TE.left(context);
+    }
+    case okta.UserStatus.PROVISIONED: {
+      return TE.left(
+        `${context} To transition user to ${activeStatus} status, please follow through with the activation workflow.`
+      );
+    }
+
+    case okta.UserStatus.LOCKED_OUT: {
+      return TE.left(
+        `${context} To transition user to ${activeStatus} status, please use the unlock command.`
+      );
+    }
+    case okta.UserStatus.PASSWORD_EXPIRED: {
+      return TE.left(
+        `${context} To transition user to ${activeStatus} status, please instruct user to login with temporary password and follow the password reset process.`
+      );
+    }
+    case okta.UserStatus.RECOVERY: {
+      return TE.left(
+        `${context} To transition user to ${activeStatus} status, please follow through with the activation workflow or restart the workflow using the reactivate-user command.`
+      );
+    }
+    case okta.UserStatus.SUSPENDED: {
+      return TE.left(
+        `${context} To transition user to ${activeStatus} status, please use the unsuspend-user command.`
+      );
+    }
+    // STAGED or DEPROVISIONED user status
+    default: {
+      return TE.right({
+        ...user,
+        status: userStatus,
+      });
+    }
+  }
+};
+
+/**
+ * Prints out what would happen if we were to activate the user.
+ * @param user - the user to dry run the activation for.
+ * @returns a TaskEither that resolves to the user.
+ */
+const dryRunActivateUser = (
+  user: ActivatableUser
+): TE.TaskEither<string, User> => {
+  // eslint-disable-next-line functional/no-expression-statement
+  console.info(
```

Comment:
**Issue:** this **must** be a `Console` to run at the 'right' time. Let me know if we need to discuss this.

Reply:
modified in commit: e743c8258d8660e8602d291c16dd3af5b11fbc97